eng/common/scripts/stress-testing/find-all-stress-packages.ps1

```diff
@@ -62,6 +80,50 @@ function GetUsername() {
     return $stressUser.ToLower()
 }
 
+function ScenariosMatrixGeneration(
+    [string]$matrixFilePath,
+    [string]$Selection,
+    [Parameter(Mandatory=$False)][string]$DisplayNameFilter,
+    [Parameter(Mandatory=$False)][array]$Filters,
+    [Parameter(Mandatory=$False)][array]$Replace,
+    [Parameter(Mandatory=$False)][array]$NonSparseParameters
+) {
+    $yamlConfig = Get-Content $matrixFilePath -Raw
+
+    $prettySerializedMatrix = &"$PSScriptRoot/../job-matrix/Create-JobMatrix.ps1" `
+        -ConfigPath $matrixFilePath `
+        -Selection $Selection `
+        -DisplayNameFilter $DisplayNameFilter `
+        -Filters $Filters `
+        -Replace $Replace `
+        -NonSparseParameters $NonSparseParameters
+
+    $prettyMatrix = $prettySerializedMatrix | ConvertFrom-Json -AsHashtable
+
+    $scenariosMatrix = @()
+    foreach($permutation in $prettyMatrix.GetEnumerator()) {
+        $entry = @{}
+        $entry.Name = $permutation.key -replace '_', '-'
+        $entry.Scenario = $entry.Name
+        $entry.Remove("Name")
+        foreach ($param in $permutation.value.GetEnumerator()) {
+            $entry.add($param.key, $param.value)
+        }
+        $scenariosMatrix += $entry
+    }
+
+    $valuesYaml = Get-Content (Join-Path $matrixFilePath '../values.yaml') -Raw
```

Comment:
What is `../values.yaml`? Does this end up being like `foo/bar/matrix.yaml/../values.yaml`? I think would be better to use split-path here.

```suggestion
    $valuesYaml = Get-Content -Raw (Join-Path (Split-Path $matrixFilePath) 'values.yaml')
```